### PR TITLE
Add poc Hedgey Strategy

### PR DESCRIPTION
**Pull request by TheShobo**
This adds a poc strategy (RFP Committeee) using Hedgey's linear vesting contracts for distribution.

One notable change is to `core/libraries/Transfer.sol` where we have set the transfer methods to `virtual`. This makes it far easier for integrations such as Hedgey to modify the base strategies and only have to override the necessary transfer method(s) instead of having to duplicate the `distribute` method and ensure it's kept up-to-date with the original.

### CODE REVIEW

**Comment by codenamejason** [comment]
I approved the action but the tests are failing.

**Comment by TheShobo** [comment]
> I approved the action but the tests are failing.

All resolved, I had missed running `forge fmt` to match the expected formatting.

**Comment by codenamejason** [comment]
> > I approved the action but the tests are failing.
> 
> All resolved, I had missed running `forge fmt` to match the expected formatting.

Happens to me all the time 🫣

**Review by thelostone-mc — DISMISSED** [review]

File: contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    /// ===============================
+    /// ======= External/Custom =======
+    /// ===============================
+
+    /// @notice Update the default Admin wallet used when creating Hedgey plans
+    /// @param _adminAddress The admin wallet to use
+    function setAdminAddress(address _adminAddress) external onlyPoolManager(msg.sender) {
+        adminAddress = _adminAddress;
```
Comment (0xKurt) [inline contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol:80]: event missing

File: contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    }
+
+    /// @notice Update the default Admin wallet used when creating Hedgey plans
+    /// @param _adminTransferOBO Set if the admin is allowed to transfer on behalf of the recipient
+    function setAdminTransferOBO(bool _adminTransferOBO) external onlyPoolManager(msg.sender) {
+        adminTransferOBO = _adminTransferOBO;
```
Comment (0xKurt) [inline contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol:86]: event missing

File: contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+    function getRecipientLockupTerm(address _recipient) external view returns (uint256) {
+        return _recipientLockupTerm[_recipient];
+    }
+
+    /// ====================================
+    /// ============ Internal ==============
+    /// ====================================
+
+    function _transferAmount(address _token, address _recipient, uint256 _amount) internal override {
```
Comment (0xKurt) [inline contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol:97]: overwriting `_transferAmount` also affects the `withdraw` function.

File: contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol (hunk 1 above)
Comment (0xKurt) [inline contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol:80]: event missing

File: contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol (hunk 2 above)
Comment (0xKurt) [inline contracts/strategies/_poc/hedgey/HedgeyRFPCommitteeStrategy.sol:86]: event missing

**Review by 0xKurt — CHANGES_REQUESTED** [review]

**Comment by 0xKurt** [comment]
I suggest to not merge the strategy until the current audit is finished because we are also auditing the RFP strategies this contract relies on, which may undergo changes.

**Comment by TheShobo** [comment]
This branch has now been updated to work with the post-audit contracts. 
@0xKurt @codenamejason

**Review by thelostone-mc — APPROVED** [review]

**Pull request merged**